src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+     *
+     * @return The default auto-commit mode.
+     * @see Builder#autoCommit(boolean)
+     */
+    public boolean isAutoCommit() {
+        return autoCommit;
+    }
+
+    /**
+     * Creates a new {@link MongoDialectSettings.Builder} based on {@code configProperties}.
+     *
+     * @param configProperties The {@linkplain Configurable#configure(Map) configuration properties}.
+     * @return A new {@link MongoDialectSettings.Builder}.
+     */
+    public static MongoDialectSettings.Builder builder(Map<String, Object> configProperties) {
+        return new MongoDialectSettings.Builder(notNull("configProperties", configProperties));
+    }
+
+    /** A builder for {@code MongoDialectSettings}. */
+    public static final class Builder {
+        private final MongoClientSettings.Builder mongoClientSettingsBuilder;
+        private @Nullable String databaseName;
+        private boolean autoCommit;
+
+        private Builder(Map<String, Object> configProperties) {
+            mongoClientSettingsBuilder = MongoClientSettings.builder();
+            var connectionString = ConfigPropertiesParser.getConnectionString(configProperties);
+            if (connectionString != null) {
+                mongoClientSettingsBuilder.applyConnectionString(connectionString);
+                databaseName = connectionString.getDatabase();
+            }
+            var autoCommit = ConfigPropertiesParser.getAutoCommit(configProperties);
+            if (autoCommit != null) {
+                this.autoCommit = autoCommit;
+            }
+        }
+
+        /**
+         * Configures {@link MongoDialectSettings}.
+         *
+         * <p>Note that if you {@link MongoClientSettings.Builder#applyConnectionString(ConnectionString)} with
+         * {@linkplain ConnectionString#getDatabase() database name} configured, you still must configure the database
+         * name via {@link Builder#databaseName(String)}, as there is no way for that to happen automatically.
+         *
+         * @param configurator The {@link Consumer} of the {@link MongoClientSettings.Builder}.
+         * @return {@code this}.
+         * @see MongoDialectSettings#getMongoClientSettings()
+         */
+        public MongoDialectSettings.Builder applyToMongoClientSettings(
+                final Consumer<MongoClientSettings.Builder> configurator) {
+            notNull("configurator", configurator).accept(mongoClientSettingsBuilder);
+            return this;
+        }
+
+        /**
+         * Sets the name of a MongoDB database used as the {@linkplain Connection#getSchema() JDBC schema} of a
+         * {@linkplain Connection connection} {@linkplain MongoConnectionProvider#getConnection() obtained} from
+         * {@link MongoConnectionProvider}.
+         *
+         * @param databaseName The name of the default MongoDB database.
+         * @return {@code this}.
+         * @see MongoDialectSettings#getDatabaseName()
+         * @see DatabaseMetaData#getSchemaTerm()
```

Comment:
Do we need to emphasize the confusing `DatabaseMetadata#getSchemaTerm()`? Even in Hibernate ORM codebase this method is never used at all.